src/engine/controls/cuecontrol.cpp

```diff
@@ -878,16 +878,14 @@ void CueControl::hotcueActivatePreview(HotcueControl* pControl, double value) {
             seekAbs(position);
             m_pPlay->set(1.0);
         }
-    } else if (m_iCurrentlyPreviewingHotcues) {
-        // This is a activate release and we are previewing at least one
-        // hotcue. If this hotcue is previewing:
-        if (pControl->isPreviewing()) {
-            // Mark this hotcue as not previewing.
-            double position = pControl->getPreviewingPosition();
-            pControl->setPreviewing(false);
-            pControl->setPreviewingPosition(Cue::kNoPosition);
-
-            // If this is the last hotcue to leave preview.
+    } else if (pControl->isPreviewing()) {
+        // Mark this hotcue as not previewing.
+        double position = pControl->getPreviewingPosition();
+        pControl->setPreviewing(false);
+        pControl->setPreviewingPosition(Cue::kNoPosition);
+        if (m_iCurrentlyPreviewingHotcues) {
```

Comment:
Please use `> 0` to indicate that this is not a boolean but a counter. The name is misleading and the `i` prefix is unsuitable to reveal the actual purpose.